### PR TITLE
py-scipy: update to 1.5.1, move gfortran10 fix

### PR DESCRIPTION
**Pull request by chrstphrchvz**
#### Description
https://docs.scipy.org/doc/scipy-1.5.1/reference/release.1.5.1.html

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.5
Xcode command line tools 11.5
MacPorts GCC 9
Python 3.8.3

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by chrstphrchvz** [comment]
@michaelld Upstream states (https://docs.scipy.org/doc/scipy/reference/toolchain.html#c-compilers):

>SciPy is compatible with most modern C compilers (in particular clang).

Is there a reason MacPorts can't use recent clang versions (e.g. MacPorts' Clang 9 or 10, or Xcode) to compile scipy?

Also, would it be difficult to allow MacPorts to select a compatible already-installed compiler without requiring a variant to be specified? In my case, I have GCC 9, Clang 9/10, and Xcode 11.5 already present, but the port defaults to wanting to install GCC 10 for building scipy.

**Comment by chrstphrchvz** [comment]
The CI build failure for py27-scipy (1.2.3) appears to be a known issue when compiling with gfortran 10: https://github.com/scipy/scipy/issues/11611. No future 1.2.x releases are planned, so I would guess MacPorts is left to backport the fix themselves or avoid using gfortran > 10.

**Comment by reneeotten** [comment]
> The CI build failure for py27-scipy (1.2.3) appears to be a known issue when compiling with gfortran 10: [scipy/scipy#11611](https://github.com/scipy/scipy/issues/11611). No future 1.2.x releases are planned, so I would guess MacPorts is left to backport the fix themselves or avoid using gfortran ≥ 10.

this build failure can likely be avoided by always using `compilers.allow_arguments_mismatch yes` (right now it's only in an `if... else...` statement). Or, if this is fixed in the newer version that flag should perhaps *not* be applied anymore, but instead only added to the PY27 subport.

**Comment by chrstphrchvz** [comment]
> this build failure can likely be avoided by always using `compilers.allow_arguments_mismatch yes` (right now it's only in an `if... else...` statement). Or, if this is fixed in the newer version that flag should perhaps _not_ be applied anymore, but instead only added to the PY27 subport.

It would also need to remain applied to the py35 subport since that is pinned at 1.4.1, and upstream only addressed it beginning in 1.5.0. What would be the best approach? Or should it not be addressed by this PR?

**Comment by chrstphrchvz** [comment]
I have moved the workaround to apply only to 1.2.3 and 1.4.1; this appears to work.

Someone had left a comment about py27-scipy 1.2.3 + gfortran10 failure on the closed ticket: https://trac.macports.org/ticket/60520#comment:6

File: python/py-scipy/Portfile [hunk 1]
```diff
@@ -60,8 +62,6 @@ if {${name} ne ${subport}} {
         # Work around thread local compiler selection bug. Remove after
         # https://github.com/macports/macports-base/pull/161 is released.
         compiler.blacklist-append {clang < 800}
-
-        compilers.allow_arguments_mismatch yes
```
Comment (michaelld) [inline python/py-scipy/Portfile:64]: Won't removing this undo the fix for https://trac.macports.org/ticket/60520 ? I think this just needs to be moved for global SciPy use.

File: python/py-scipy/Portfile [hunk 2]
```diff
@@ -7,10 +7,10 @@ PortGroup               github 1.0
 PortGroup               compilers 1.0
 PortGroup               compiler_blacklist_versions 1.0
 
-github.setup            scipy scipy 1.5.0 v
-checksums               rmd160  bae805468c0781ea8fff87f25b7c3384dcd0875e \
-                        sha256  b31b616b4571f19a4c1a925b1b19f4eb325d098265692d88ea386537aa3742ab \
-                        size    19831004
+github.setup            scipy scipy 1.5.1 v
+checksums               rmd160  112cc52964a9c7890328ea53045bfac024309f7f \
```
Comment (michaelld) [inline python/py-scipy/Portfile:11]: Let's move this to a separate commit from Fortran tweaks.

File: python/py-scipy/Portfile (hunk 1 above)
Comment (reneeotten) [reply python/py-scipy/Portfile:64]: upstream presumably [resolved](https://github.com/scipy/scipy/pull/11842) this issue in 1.5.0 (?); if that's indeed the case it should only be needed for the PY27/PY35 subports that are pinned to an earlier version.

File: python/py-scipy/Portfile (hunk 2 above)
Comment (chrstphrchvz) [reply python/py-scipy/Portfile:11]: Done.

File: python/py-scipy/Portfile (hunk 1 above)
Comment (chrstphrchvz) [reply python/py-scipy/Portfile:64]: > upstream presumably [resolved](https://github.com/scipy/scipy/pull/11842) this issue in 1.5.0 (?)

This is according to https://docs.scipy.org/doc/scipy/reference/release.1.5.0.html:
>## Issues closed for 1.5.0
> * \#11611: Build failure with gfortran 10
>## Pull requests for 1.5.0
> * \#11842: BLD: fix build with gfortran 10

Comment (chrstphrchvz) [reply python/py-scipy/Portfile:64]: I agree that removing the fix for the latest version is likely a matter of preference, at least right now. I don't know if upstream will eventually rework the affected code and will want scipy to be built without this flag; even then, it would presumably not hurt for it to be set anyway by MacPorts.

Comment (michaelld) [reply python/py-scipy/Portfile:64]: Just tested & it looks like SciPy 1.5.0+ builds cleanly without this Fortran tweak ... so, looks like it's just needed for older versions. yay!
